pydantic/main.py

```diff
@@ -126,6 +127,17 @@ def hash_function(self_: Any) -> int:
 # the `BaseModel` class, since that's defined immediately after the metaclass.
 _is_base_model_class_defined = False
 
+JSON_UNENCODABLE_TYPES: Tuple[Type[Any], ...] = (
```

Comment:
This name is extremely confusing, surely it should be called `JSON_TYPES`, also:
* it needs a comment explaining how it's used
* could we move this and its love into `json.py`.